presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -60,9 +65,30 @@ public Connection openConnection(JdbcIdentity identity)
             throws SQLException
     {
         Properties properties = getCredentialProperties(identity);
-        Connection connection = driver.connect(connectionUrl, properties);
-        checkState(connection != null, "Driver returned null connection");
-        return connection;
+
+        int retryCount = 0;
+        SQLException error = null;
+
+        while (retryCount <= 10) {
+            try {
+                Connection connection = driver.connect(connectionUrl, properties);
+                checkState(connection != null, "Driver returned null connection");
+                return connection;
+            }
+            catch (SQLException e) {
+                retryCount = retryCount + 1;
```

Comment:
Instead of having it here can we wrap the `DriverConnectionFactory' with a  `RetryableDriverConnectionFactory` which could capture the exception and retry it for n times ? We can configure the initial retry time and the retry count in it ?

Reply:
@Praveen2112 I thought we can add properties to `BaseJdbcConfig` for making the retry behavior configurable. However, if we will create `RetryableDriverConnectionFactory` aside from `DriverConnectionFactory`, it may not be a good idea?

Also, should all JDBC connectors be updated to use `RetryableDriverConnectionFactory` if it will be available in base-jdbc module?